cli/cycpp.py

```diff
@@ -1986,39 +1984,45 @@ def outter_split(s, open_brace='(', close_brace=')', separator=','):
             val += separator
     return outter
 
-def split_template_args(s, open_brace='<', close_brace='>', separator=','):
-    """Takes a string with template specialization and returns a list
-    of the argument values as strings. Mostly cribbed from xdress.
-    """
-    targs = []
-    ns = s.split(open_brace, 1)[-1].rsplit(close_brace, 1)[0].split(separator)
-    count = 0
-    targ_name = ''
-    for n in ns:
-        count += int(open_brace in n)
-        count -= int(close_brace in n)
-        if len(targ_name) > 0:
-            targ_name += separator
-        targ_name += n
-        if count == 0:
-            targs.append(targ_name.strip())
-            targ_name = ''
-    return targs
-
 def parse_template(s, open_brace='<', close_brace='>', separator=','):
     """Takes a string -- which may represent a template specialization --
-    and returns the corresponding type. Mostly cribbed from xdress.
+    and returns the corresponding type.
     """
-    if open_brace not in s and close_brace not in s:
+    s = s.replace(' ', '')
+    if open_brace not in s and separator not in s:
         return s
-    t = [s.split(open_brace, 1)[0]]
-    targs = split_template_args(s, open_brace=open_brace,
-                                close_brace=close_brace, separator=separator)
-    for targ in targs:
-        t.append(parse_template(targ, open_brace=open_brace,
-                                close_brace=close_brace, separator=separator))
+
+    i = s.find(open_brace)
+    j = s.rfind(close_brace)
+    t = [s[:i]]
+    inner = s[i+1:j]
+    t.extend(parse_arg(inner, open_brace, close_brace, separator))
     return t
 
+def parse_arg(s, open_brace, close_brace, separator):
+    """Takes a string containing one or more c++ template args and returns
+    a list of the argument values as strings.
+    """
```

Comment:
Do you mean just in the docstring?

On Mon, Mar 30, 2015 at 3:05 PM, Matthew Gidden <notifications@github.com>
wrote:

> In cli/cycpp.py
> <https://github.com/cyclus/cyclus/pull/1108#discussion_r27431854>:
>
> >      return t
> >
> > +def parse_arg(s, open_brace, close_brace, separator):
> > +    """Takes a string containing one or more c++ template args and returns
> > +    a list of the argument values as strings.
> > +    """
>
> (please)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cyclus/pull/1108/files#r27431854>.
>